src/pyproject_fmt/formatter/pep508.py

```diff
@@ -10,36 +12,19 @@
 REQ_REGEX = re.compile(r"(===|==|!=|~=|>=?|<=?|@)\s*([^,]+)")
 
 
-def _req_base(lib: str) -> str:
-    match = re.match(BASE_NAME_REGEX, lib)
-    if match is None:
-        raise ValueError(repr(lib))
-    return match.group(0)
-
-
-def _normalize_lib(lib: str) -> str:
-    base = _req_base(lib)
-    values = sorted(
-        (f"{m.group(1)}{m.group(2)}" for m in REQ_REGEX.finditer(lib)), key=lambda c: ("<" in c, ">" in "c", c)
-    )
+def normalize_req(req: str) -> str:
+    parsed = Requirement(req)
+    orig_specifiers = str(parsed.specifier)
+    values = [f"{m.group(1)}{m.group(2)}" for m in REQ_REGEX.finditer(orig_specifiers)]
     if values:  # strip .0 version
         while values[0].endswith(".0") and (values[0].startswith(">=") or values[0].startswith("==")):
             values[0] = values[0][:-2]
-    return f"{base}{','.join(values)}"
```

Comment:
In the tests with examples from the PEP 508 tests, I noticed that `_normalize_lib` was having problems with extras, so I rewrote the function to rely on `packaging.requirements.Requirement` parsing to extract the specifiers set.

Since `packaging.requirements.Requirement` already sorts the specifier set, I removed the sorting code, but I kept the `.0` normalisation, to preserve the original intentions of the code.

Reply:
Pushed a commit that incorporates my feedback :+1:

Reply:
Thank you very much for the feedback and improvements.